README.md

```diff
@@ -120,6 +120,22 @@ in
 }
 ```
 
+### Shell Autocompletion
+
+#### Nushell
+
+Run this command for each new version of bake:
+
+```text
+bake --completions nushell | save -f ~/.config/nushell/completions-bake.nu
+```
+
+And then include this in your `~/.config/nushell/config.nu` file:
+
+```text
+source completions-bake.nu
```

Comment:
The generated file is a module, so we should be able to avoid clobbering the environment more than we have to.

```suggestion
overlay use completions-bake.nu
```

Reply:
All other tools including the docs use the same technique of `sourcing` and I think I also like it better than having an overlay.

Reply:
I've moved the completion to a module and hope that this helps with the namespace cluttering. Not sure though.

I prefer this to be consistent with all the other completions and not use an overlay.

Reply:
This is "just" a documentation/local configuration thing (aside from using `export` correctly, which it already did). A module is just a file, just like in Rust. To import it as a snippet, use `source`, to import it as a module, use `[overlay] use`.

The explicit `module` command is only for submodules.

Reply:
I'm afraid I don't understand that comment. Can you help me by rephrasing? And remember, I'm new to NuShell so you might have to go back to the basics.

What are you referring to with "This" for example?

I basically copied and pasted what I'm seeing from other completions, they can't be all wrong?

e.g.
https://github.com/nushell/nu_scripts/blob/main/custom-completions/git/git-completions.nu
https://github.com/nushell/nu_scripts/tree/main/custom-completions

Reply:
You have three ways to import another source file in Nu:

- `source file.nu` (include): all "things" (commands, externs, etc; not sure if there is a better name for this) are imported into your current scope (like Rust's `include!("file.rs")`)
- `use file.nu` (module): only things marked `export` are imported into your current scope (like Rust's `use file::*;` and `pub`, respectively), things in the imported module can still use its own private commands
- `overlay use file.nu` (overlay): like a module, but can be disabled by `overlay hide file`

As a module author, things look the same to you in all three cases. The only thing you need to do is annotate the things you intend the user to use as `export def`/`export extern`, which you already do.

The example you link does the same thing: `extern`s are `export`ed, private utility functions are not. The documentation (https://github.com/nushell/nu_scripts/tree/main/custom-completions#custom-completions) also mentions that both `source` and `use` are supported.

Reply:
So, that means I do not clutter the namespace after all and all is good on this front?